hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java

```diff
@@ -103,7 +109,12 @@ public boolean getBoolean(HazelcastProperty groupProperty) {
      * @throws NumberFormatException if the value cannot be parsed
      */
     public int getInteger(HazelcastProperty groupProperty) {
-        return Integer.parseInt(properties[groupProperty.getIndex()]);
+        int configuredValue = Integer.parseInt(properties[groupProperty.getIndex()]);
+        if (configuredValue < 0) {
```

Comment:
Since we have at least one property where there is a negative default value, we should maybe check this here. We could have another field in GroupProperty to allow negative values or check if the default is negative.